src/tiled/gotodialog.cpp

```diff
@@ -0,0 +1,94 @@
+#include "gotodialog.h"
+
+#include <QApplication>
+#include <QHBoxLayout>
+#include <QGroupBox>
+#include <QLabel>
+#include <QPushButton>
+#include <QIntValidator>
+#include <QString>
+#include "documentmanager.h"
+#include "mapview.h"
+#include "mapscene.h"
+#include "mapreader.h"
+#include "mapdocument.h"
+#include "maprenderer.h"
+
+using namespace Tiled::Internal;
+using namespace Tiled;
+
+GotoDialog *GotoDialog::mInstance;
+
+GotoDialog::GotoDialog(QWidget *parent, Qt::WindowFlags f )
```

Comment:
It should be called the `GotoTileDialog` unless we have bigger plans for it. Actually, I think it would be great if it could also jump to named objects, so we can keep this name for now.